src/DBAL/EnumType.php

```diff
@@ -63,7 +63,7 @@ public function isEnumNameTypeMapping(): bool
     /**
      * {@inheritdoc}
      */
-    public function convertToPHPValue($value, AbstractPlatform $platform)
+    public function convertToPHPValue($value, AbstractPlatform $platform): ?Enum
```

Comment:
I'm not sure whether this work if `doctrine/dbal` project introduce signatures on their classes at some point. I think we should avoid this at the moment for interfaces\overrides\etc

Reply:
They already did: it is already marked as "mixed" in 3.3
https://github.com/doctrine/dbal/blob/3.3.x/src/Types/Type.php#L83 in PHPDoc typehint
and it will be real mixed in 4.x:
https://github.com/doctrine/dbal/blob/4.0.x/src/Types/Type.php

Reply:
Good point. Did a verification test, looks OK https://3v4l.org/DLWKM